src/libraries/System.Net.Mail/src/System/Net/Mail/MailAddressCollection.cs

```diff
@@ -51,25 +51,26 @@ internal void ParseValue(string addresses)
 
         internal string Encode(int charsConsumed, bool allowUnicode)
         {
-            string encodedAddresses = string.Empty;
+            StringBuilder? encodedAddresses = null;
```

Comment:
> There is a `ValueStringBuilder` for BCL, but S.N.Mail isn't using it.

Yeah, `System.Net.Mail` has a quite large field for improvements.

> `MailAddress.Encode` also has usage of string concatenation.

That's intentionally out of scope for not to make a huge PR having lots of changes in many places. In addition to that currently I do not have the same amount of time I had before to spend on .NET contributions as before, so tiny changes are more preferable for me at the moment.

Reply:
> Yeah, System.Net.Mail has a quite large field for improvements.

System.Net.Mail is not actively maintained because the main type SmtpClient is obsolete. See remark at https://learn.microsoft.com/en-us/dotnet/api/system.net.mail.smtpclient?view=net-8.0.